src/tests/java/htsjdk/samtools/sra/SRATest.java

```diff
@@ -0,0 +1,551 @@
+/*===========================================================================
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.*;
+
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.reference.ReferenceSequenceFile;
+import htsjdk.samtools.sra.SRAAccession;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+/**
+ * Created by andrii.nikitiuk on 8/24/15.
+ */
+public class SRATest {
+
+    @DataProvider(name = "testCounts")
+    public Object[][] createDataForCounts() {
+        return new Object[][] {
+            {"SRR2096940", 10591, 498}
+        };
+    }
+
+    @Test(dataProvider = "testCounts")
+    public void testCounts(String acc, int numberAlignments, int numberUnalignments) {
+        if (!SRAAccession.isSupported()) return;
```

Comment:
```
    [javac] /home/nikitiuk/work/ncbi/misc/htsjdk/src/tests/java/htsjdk/samtools/sra/SRATest.java:35: error: cannot find symbol
    [javac] import org.testng.SkipException;
    [javac]                  ^
    [javac]   symbol:   class SkipException
    [javac]   location: package org.testng
```

I think that is because HTSJDK uses old version of testng. It only allows to use annotation to disable tests, but that can't be done dynamically